packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/EnumTypeProvider.cs

```diff
@@ -0,0 +1,363 @@
+                };
+
+                _typeMapping = sourceInputModel?.CreateForModel(ExistingType);
+            }
+
+            if (isExtensible)
+            {
+                // extensible enums are implemented by readonly structs
+                DeclarationModifiers |= TypeSignatureModifiers.Partial | TypeSignatureModifiers.ReadOnly;
+            }
+
+            Name = input.Name.ToCleanName();
+            Namespace = GetDefaultModelNamespace(CodeModelPlugin.Instance.Configuration.Namespace);
+            IsExtensible = isExtensible;
+            ValueType = CodeModelPlugin.Instance.TypeFactory.CreateCSharpType(input.EnumValueType);
+            IsStringValueType = ValueType.Equals(typeof(string));
+            IsIntValueType = ValueType.Equals(typeof(int)) || ValueType.Equals(typeof(long));
+            IsFloatValueType = ValueType.Equals(typeof(float)) || ValueType.Equals(typeof(double));
+            IsNumericValueType = IsIntValueType || IsFloatValueType;
+
+            Description = input.Description;
+
+            _valueField = new FieldDeclaration(FieldModifiers.Private | FieldModifiers.ReadOnly, ValueType, "_value");
+
+            if (!IsExtensible)
+            {
+                SerializationProviders = CodeModelPlugin.Instance.GetSerializationTypeProviders(this);
+            }
+        }
+
+        private readonly FieldDeclaration _valueField;
+
+        public CSharpType ValueType { get; }
+        public bool IsExtensible { get; }
+        public bool IsIntValueType { get; }
+        internal bool IsFloatValueType { get; }
+        public bool IsStringValueType { get; }
+        internal bool IsNumericValueType { get; }
+        public string? Description { get; }
+        public override string Name { get; }
+        public override string Namespace { get; }
+        protected override TypeKind TypeKind => IsExtensible ? TypeKind.Struct : TypeKind.Enum;
+
+        /// <summary>
+        /// The serializations providers for the model provider.
+        /// </summary>
+        public IReadOnlyList<TypeProvider> SerializationProviders { get; } = Array.Empty<TypeProvider>();
+
+        private IReadOnlyDictionary<EnumTypeValue, FieldDeclaration>? _valueFields;
+        private IReadOnlyDictionary<EnumTypeValue, FieldDeclaration> ValueFields => _valueFields ??= BuildValueFields();
+
+        private IReadOnlyList<EnumTypeValue>? _values;
+        public IReadOnlyList<EnumTypeValue> Values => _values ??= BuildValues();
+
+        private IReadOnlyList<EnumTypeValue> BuildValues()
+        {
+            var values = new EnumTypeValue[_allowedValues.Count];
+
+            for (int i = 0; i < values.Length; i++)
+            {
+                values[i] = new EnumTypeValue(_allowedValues[i]);
+            }
+
+            return values;
+        }
+
+        public bool TryGetFieldFromValue(EnumTypeValue value, [MaybeNullWhen(false)] out FieldDeclaration field)
```

Comment:
Does this need to be public?

Reply:
No, I changed this to private because I changed the enum serialization type provider to a nested class of `EnumTypeProvider` because it needs quite a few things from the enum